src/libc/errno.rs

```diff
@@ -5,6 +5,37 @@
  */
 //! `errno.h`
 
+use crate::dyld::FunctionExports;
+use crate::environment::Environment;
+use crate::export_c_func;
+use crate::mem::MutPtr;
+
 pub const EPERM: i32 = 1;
 pub const EDEADLK: i32 = 11;
 pub const EINVAL: i32 = 22;
+
+#[derive(Default)]
+pub struct State {
+    errnos: std::collections::HashMap<crate::ThreadID, MutPtr<i32>>,
+}
+impl State {
+    fn errno_for_thread(
+        &mut self,
+        mem: &mut crate::mem::Mem,
+        thread: crate::ThreadID,
+    ) -> MutPtr<i32> {
+        // TODO: "real" errno implementation
+        *self
+            .errnos
+            .entry(thread)
+            .or_insert_with(|| mem.alloc_and_write(0i32))
+    }
+}
+
+fn __error(env: &mut Environment) -> MutPtr<i32> {
+    env.libc_state
```

Comment:
Can you add something like `log!("TODO: errno (returned 0)");`? If we're always returning `0`, I'm sure _something_ will break, and it'd be good to at least have a warning about it in the log.

Reply:
I can, but the problem is that it become very spammy - as Wolf3d loads a new chunk of ogg on each frame, it also check errno value :/

I can use log_dbg!(), is it acceptable ?

Reply:
I checked that was done for time.rs, but I'm not a big fan of introducing a local state just to make logging better

Reply:
I don't really want to use `log_dbg!` for this kind of thing, because there's no hint that it's used…

But I just realised, we don't need any new state! You are already creating the errno variable on demand, you can put the `log!` in there (`log!("TODO: errno accessed on thread {} (will always be 0)");`)

Reply:
Right, done